.ci/Jenkinsfile

```diff
@@ -58,18 +59,57 @@ pipeline {
               from: "${env.GIT_PREVIOUS_SUCCESSFUL_COMMIT}",
               to: "${env.GIT_COMMIT}",
               patterns: regexps)
+
+            // regexp for JSON specs
+            regexps =[
+              "^tests/agents/json-specs/.*"
+            ]
+            env.JSON_SPECS_UPDATED = isGitRegionMatch(
+              from: "${env.GIT_PREVIOUS_SUCCESSFUL_COMMIT}",
+              to: "${env.GIT_COMMIT}",
+              patterns: regexps)
           }
         }
         stash allowEmpty: true, name: 'source', useDefaultExcludes: false
       }
     }
-    stage('Send Pull Request'){
+    stage('Send Pull Request for BDD specs'){
+      options {
+        warnError('Pull Requests to APM agents failed')
+      }
+      when {
+        beforeAgent true
+        anyOf {
+          expression { return env.GHERKIN_SPECS_UPDATED != "false" }
+          expression { return params.FORCE_SEND_PR == "true" }
```

Comment:
NIT
```suggestion
          expression { return params.FORCE_SEND_PR }
```

Reply:
I'm not sure if this works. Groovy truth is a little strange sometimes. IIRC, if the return value is `boolean` and you return a String, I think Groovy converts the String to `true` if the string is non-empty. So please test that if the value is `"false"`, it actually returns `false`. Maybe Jenkins already converts a `"false"` to `false` though. If not, you can do `params.FORCE_SEND_PR.toBoolean()`.

Reply:
Yes, I'm currently looking at the expected behavior, as I remember having previous one for this reason.

FYI I'm elaborating a new PR here: https://github.com/elastic/apm/pull/320